solr/core/src/test/org/apache/solr/handler/JsonLoaderTest.java

```diff
@@ -946,5 +946,87 @@ public void testGrandChildDocs() throws Exception {
 
   }
 
+  @Test
+  public void testRelationalChildDocs() throws Exception {
```

Comment:
A bit on terminology.... if so-called "relational" child docs are going to be the only way to have a child doc in the future, then lets just start calling these simply child docs without the "relational" qualifier.  "Relational" may be kinda confusing since an anonymous relationship is still related.  Perhaps the new way is a "labelled" child doc?  If you disagree and want to keep some qualifier then okay.  If you see an existing test that says "child doc" for anonymous child doc then change its name to be about "anonymous" child docs.